### PR TITLE
Throw a more informative exception if reference lookup failed

### PR DESCRIPTION
**Pull request by pityka**
The original version may passed a null pointer to `restoreReadBases`. This throws the same exception as in CRAMIterator.

### CODE REVIEW

**Comment by yfarjoun** [comment]
can you add a test for this simple case?

**Comment by pityka** [comment]
I have a file for which CRAMIterator throws this exception (NPE in master). I tried to isolate the read to make a test case and I noticed that `samtools view` hangs halfway upon reading the file, so that file might be simply truncated or otherwise wrong. 
Unfortunately I don't know how to create a cram file from scratch which would test this.

**Comment by cmnbroad** [comment]
There is a discussion ongoing in #362 about how to tighten up the validation and error checking for the CRAM code.

File: src/java/htsjdk/samtools/cram/build/CramNormalizer.java [hunk 1]
```diff
@@ -123,9 +124,15 @@ public void normalize(final ArrayList<CramCompressionRecord> records,
...
                 record.readBases = SAMRecord.NULL_SEQUENCE;
-            } else
+            } else {
+                if (refBases != null) {
                 record.readBases = restoreReadBases(record, refBases, refOffset_zeroBased,
                         substitutionMatrix);
+                } else {
+                     throw new CRAMException(String.format("Contig %s not found in the reference file.", header.getSequence(record.sequenceId).toString()));
+                }
+            }
```
Comment (nh13) [inline src/java/htsjdk/samtools/cram/build/CramNormalizer.java]: why not just `else if { ...} else { ... }` instead of nesting an `if/else` inside an `else`?

Comment (nh13) [inline src/java/htsjdk/samtools/cram/build/CramNormalizer.java]: Also, formatting.

**Comment by akiezun** [comment]
agreed, needs a test. looks fine otherwise

**Comment by vadimzalunin** [comment]
Seems ok to me. 
@pityka could you please add a test? A fake record with an empty ref source should do it.

**Comment by droazen** [comment]
Since @pityka has not responded, @vadimzalunin has agreed to incorporate this fix into an upcoming branch, with a test.
